python/packages/nisar/products/readers/SLC/RSLC.py

```diff
@@ -84,6 +85,39 @@ def getSlcDatasetAsNativeComplex(
         else:
             return dataset
 
+    def isSlcDatasetReceiveOnly(
+            self,
+            frequency: str,
+            polarization: str):
+        '''
+        Returns if an SLC dataset has been acquired in "receive-only"
+        mode
+
+        Parameters
+        ----------
+        freq : "A" or "B"
+            The frequency letter, either "A" or "B".
+        pol: str
+            The polarization term associated with the SLC dataset
+
+        Returns
+        -------
+        is_receive_only: bool or None
+            Flag indicating whether the SLC has been acquired in
+            receive-only mode, or None if undetermined.
+        '''
+        slc_pol_dataset = self.getSlcDatasetAsNativeComplex(
```

Comment:
Here we should use the [`getSlcDataset()`](https://github.com/isce-framework/isce3/blob/675bde1e32611f856606d11f95d8654b70fe1ac1/python/packages/nisar/products/readers/SLC/SLCBase.py#L9) method that always returns an `h5py.Dataset` object.

The [`getSlcDatasetAsNativeComplex()`](https://github.com/isce-framework/isce3/blob/675bde1e32611f856606d11f95d8654b70fe1ac1/python/packages/nisar/products/readers/SLC/RSLC.py#L52-L56) method might return a [`ComplexFloat16Decoder`](https://github.com/isce-framework/isce3/blob/675bde1e32611f856606d11f95d8654b70fe1ac1/python/packages/isce3/core/types.py#L99) object instead if the RSLC product used half-precision. `ComplexFloat16Decoder` objects don't have `attrs` so we can't use them here.

```suggestion
        slc_pol_dataset = self.getSlcDataset(
```